fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java

```diff
@@ -742,7 +742,7 @@ private Void computeJoinNode(ExpressionContext context, ScalarOperator joinOnPre
                 builder.setOutputRowCount(max(0, rightRowCount - innerRowCount));
                 break;
             case FULL_OUTER_JOIN:
-                builder.setOutputRowCount(leftRowCount + rightRowCount - innerRowCount);
+                builder.setOutputRowCount(max(0, leftRowCount + rightRowCount - innerRowCount));
```

Comment:
leftRowCount + rightRowCount should always large than `innerRowCount`, The real error seems happen before

Reply:
I think max(0, leftRowCount + rightRowCount - innerRowCount) is better